In [1]:
import pandas as pd
from sqlalchemy import create_engine

In [2]:
db_url = "postgresql+psycopg2://postgres:abc123@localhost/prosperous_universe"
engine = create_engine(db_url)

In [3]:
query = """
    SELECT *
    FROM raw.recipe_report;
"""

In [4]:
with engine.connect() as connection:
    df = pd.read_sql_query(query, con=connection)

In [5]:
df

,original_query,prefix,Output_Material,volume,input_materials_AI1_AskPrice_per_unit,total input_materials_AI1_AskPrice,total output_materials_AI1_BidPrice,profit of buy minus sell,profit per day,traded_rank
0,SME:6xALO-1xO-1xC-1xFLX=>4xAL,SME,AL,4.0,889.500000,3558.0,5200.0,1642.0,2736.666667,1
1,SME:6xALO-1xC-1xO=>3xAL,SME,AL,3.0,995.333333,2986.0,3900.0,914.0,1828.000000,1
2,SME:1xSCR-1xO=>6xS,SME,S,6.0,169.333333,1016.0,1200.0,184.0,613.333333,8
3,SME:5xCUO-10xO-1xSIO=>3xCU,SME,CU,3.0,1789.333333,5368.0,5430.0,62.0,124.000000,7
4,SME:1xC-3xAUO=>2xAU,SME,AU,2.0,2945.000000,5890.0,5400.0,-490.0,-1088.888889,9
5,SME:6xFEO-1xC-1xO-1xFLX=>4xFE,SME,FE,4.0,1318.500000,5274.0,4200.0,-1074.0,-1790.000000,4
6,SME:6xFEO-1xC-1xO=>3xFE,SME,FE,3.0,1567.333333,4702.0,3150.0,-1552.0,-3104.000000,4
7,SME:10xLIO-4xHAL=>4xLI,SME,LI,4.0,2608.000000,10432.0,7360.0,-3072.0,-4654.545455,3
8,SME:4xTIO-1xC-1xO=>2xTI,SME,TI,2.0,4293.000000,8586.0,4100.0,-4486.0,-5981.333333,6
9,SME:3xSIO-1xC-1xO=>1xSI,SME,SI,1.0,3640.000000,3640.0,2350.0,-1290.0,-6450.000000,2


In [6]:
# Use method='spearman' for ordinal rank data
corr = df['profit per day'].corr(df['traded_rank'], method='spearman')
print(f"Spearman correlation: {corr:.3f}")

Spearman correlation: -0.300


Key Finding: A weak-to-moderate negative Spearman rank correlation ($\rho = -0.300$) exists between an item's traded rank and its profit per day. Because lower rank numbers correspond to higher trading volume (Rank 1 being the most traded), this negative correlation indicates that more frequently traded items tend to generate higher daily profits. However, since the correlation is moderate ($\rho = -0.300$), trade volume alone is not the sole driver of profitability;